itou/openid_connect/models.py

```diff
@@ -191,15 +196,26 @@ def create_or_update_user(self, is_login=False):
         self.check_valid_kind(user, user_data_dict, is_login)
 
         if not created:
-            for key, value in user_data_dict.items():
+            user_update_dict = user_data_dict.copy()
+            readonly_pii_fields = user.jobseeker_profile.readonly_pii_fields() if user.is_job_seeker else set()
+            if readonly_pii_fields:
+                for field in readonly_pii_fields:
+                    user_update_dict.pop(field, None)
+                logger.info(
+                    "Not updating fields %s on job seeker pk=%d because their identity has been certified.",
+                    ", ".join(api_particulier.USER_REQUIRED_FIELDS),
+                    user.pk,
+                )
+            else:
+                if birthdate is not _no_birthdate and user_data_dict["kind"] == UserKind.JOB_SEEKER:
```

Comment:
Même si cela revient au même (car tous nos `IdentityCertification` certifie la date de naissance), je trouverais cela plus logique de vérifier explicitement `if "birthdate" not in readonly_pii_fields`.

En poussant le trait, on pourrait avoir `ProConnect` qui fournit une `IdentityCertification` de l'email qui rendrait le champs email readonly.